kafka/consumer/fetcher.py

```diff
@@ -478,8 +478,8 @@ def _unpack_message_set(self, tp, records):
         # caught by the generator. We want all exceptions to be raised
         # back to the user. See Issue 545
         except StopIteration as e:
-            log.exception('StopIteration raised unpacking messageset: %s', e)
-            raise Exception('StopIteration raised unpacking messageset')
+            log.exception('StopIteration raised unpacking messageset')
+            raise RuntimeError('StopIteration raised unpacking messageset')
```

Comment:
Potentially this should be a `KafkaError` although I'm unclear when to reach for that instead of `RuntimeError`

Reply:
RuntimeError is what would be raised by the convention on Python3 if we have a not handled StopIteration raised inside of a generator.